src/Parsers/ASTFunction.cpp

```diff
@@ -229,68 +229,32 @@ void ASTFunction::formatImplWithoutAlias(const FormatSettings & settings, Format
                 }
 
                 const auto * literal = arguments->children[0]->as<ASTLiteral>();
-                /* A particularly stupid case. If we have a unary minus before
```

Comment:
So I had to remove this, otherwise it makes formatting of negate inconsistent, i.e. `-(-(-1))` -> ` -(1)` -> `-1`